maint-lib/ceph_version.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+CEPH_VERSION_SPEC=$1
+WHAT_TO_EXTRACT=$2
+
+# Search for the two possible separator between CEPH_VERSION and CEPH_POINT_RELEASE
+# Let's consider CEPH_VERSION_SPEC=luminous-12.2.0-1
+for separator in "=" "-"; do
+  # If the line doesn't have a separator, let's try the next one
+  echo "${CEPH_VERSION_SPEC}" | grep -q "${separator}" || continue
```

Comment:
When possible, bash constructs should be used over outside tools and pipes. In this case, it is easier and clearer to write:
```sh
if [[ ! "${CEPH_VERSION_SPEC}" =~ "${separator}" ]]; then continue; fi
```

Reply:
We already have some code using grep. Is that much important ?

Reply:
This is part of shell scripting best practices. More pipes means more opportunity for errors to occur. Like if a particular version of grep is bugged for some reason. If the dependency on grep is removed, that is no longer a failure mode.